baseband/dada/base.py

```diff
@@ -282,10 +280,11 @@ def read(self, count=None, out=None):
             nsample = min(count, self.samples_per_frame - sample_offset)
             sample = self.offset - offset0
             data_slice = slice(sample_offset, sample_offset + nsample)
-            if self.thread_ids:
-                data_slice = (data_slice, self.thread_ids)
-
-            result[sample:sample + nsample] = self._frame[data_slice]
+            if self.subset:
+                data_slice = (data_slice,) + self.subset
+            out[sample:sample + nsample] = (
+                self._frame[data_slice].squeeze() if self.squeeze else
```

Comment:
You have two different ways of applying the `subset` -- here, you construct a slice and then use it; in some of the other formats, you slice `data` sequentially. The one here is arguably better, though perhaps less clear.

Reply:
I made the switch to this syntax in the VLBI formats, but had to modify some of the lines for VDIF and Mark 4 (the latter because Mark 4 frame's `__getitem__` forbids indexing).

Reply:
Could you raise an issue about `Mark4Frame` not allowing indexing? That may just be an oversight.